### ISSUE TITLE
Certbot-dns-google fails with HTTP 412 - Precondition not met

### ISSUE THREAD

**Issue by mowirth** [certbot/certbot#6036]
## My operating system is (include version):
CentOS 7

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
yum install -y certbot
pip install certbot-dns-google

## I ran this command and it produced this output:

certbot certonly --dns-google --dns-google-credentials <path> --server https://acme-v02.api.letsencrypt.org/directory -d ssl1234.example.net -d *.example.net -d example.net  ...

## Certbot's behavior differed from what I expected because:

Encountered error adding TXT record: <HttpError 412 when requesting https://www.googleapis.com/dns/v1/projects/projectname/managedZones/id/changes?alt=json returned "Precondition not met for 'entity.change.deletions[0]'">

This problem only appeared if multiple domains are requested (around 15 SAN Names), single Domains work fine. I tested both wildcard domains and single domains - both fail sometimes. 

###### Logs are stored in `/var/log/letsencrypt` by default. Feel free to redact domains, e-mail and IP addresses as you see fit.

2018-05-24 16:05:21,163:WARNING:certbot_dns_google.dns_google:Encountered error deleting TXT record: <HttpError 412 when requesting https://www.googleapis.com/dns/v1/projects/projectname/managedZones/id/changes?alt=json returned "Precondition not met for 'entity.change.deletions[0]'">

**Comment by rigrassm** [comment]
Was just dealing with this and it turns out the issue for me was there was an existing txt record still in my domain's zone from when I had manually done the DNS challenge option the first time around. The API call used to delete that existing record is not providing the expected parameter for entity.change.deletions[0] so it's not able to delete the old entry.

I haven't dug into what exactly is expected in that API call so not sure what parameter isn't being provided correctly but you can work around this by deleting any old TXT records in the domain's zone that were used for the DNS challenge previously.

After that, certbot is able to add the TXT record and also removes it once it finishes so it shouldn't happen again.

**Comment by mouwirth** [comment]
I think this has something to do with subdomains that we use - we have one zone for example.org and also issue *.subdomain1.example.org and *.subdomain2.example.org - this works with the root domain and one subdomain but it seems to fail with another subdomain

**Comment by bmw** [comment]
It's been almost a year since this issue has seen any activity so I'm closing this issue. If you're still having this problem, please comment or open a new issue.

**Comment by pjjw** [comment]
Hey, just ran into this myself here. I'm attempting to renew *.labs.shortbus.org which I did manually before, and it's failing with this same error.

**Comment by pjjw** [comment]
Exactly the same issue, had a TXT record left around from the last challenge. When I removed "_acme-challenge.labs.shortbus.org." the google plugin worked again, but it seems it wasn't able to remove that record on its own, or it wasn't doing it correctly.

**Comment by pjjw** [comment]
tl;dr for other lost souls ending up here: rigrassm's workaround works.

**Comment by judepereira** [comment]
Just hit it again - rigrassm's workaround works.

**Comment by bmw** [comment]
OK. I'm reopening the issue.

**Comment by rigrassm** [comment]
I did a quick dive into the module and noticed that the `TXT` record deletion is only performed during the [`_cleanup` stage.](https://github.com/certbot/certbot/blob/f8e097a06176edfd69cd7306b6c83e3946b47f60/certbot-dns-google/certbot_dns_google/dns_google.py#L72)
```
    def _cleanup(self, domain, validation_name, validation):
        self._get_google_client().del_txt_record(domain, validation_name, validation, self.ttl)
```

Since this issue is being caused by the TXT record existing prior to utilizing this module to make the changes to the zone, the `_perform` stage runs into the collision with the existing record and doesn't have the logic built in to handle that scenario.

```
    def _perform(self, domain, validation_name, validation):
        self._get_google_client().add_txt_record(domain, validation_name, validation, self.ttl)
```

I'd imagine the easiest route to resolving this issue here would be to add a check for an existing record to the `_perform` function and delete it if it finds one before moving on to adding the new record.
